### PR TITLE
Add support for mgmt api ListKeyspaces op

### PR DESCRIPTION
**Pull request by adejanovski**
**What this PR does**:
Adds support for the newly added list keyspaces operation in the management API: https://github.com/k8ssandra/management-api-for-apache-cassandra/commit/e77aa6ab32fb119fbb865cf8edc3f8a9a37eab95

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: pkg/httphelper/client.go [hunk 1]
```diff
@@ -280,6 +280,37 @@ func (client *NodeMgmtClient) modifyKeyspace(endpoint string, pod *corev1.Pod, k
...
+func parseListKeyspacesEndpointsResponseBody(body []byte) ([]string, error) {
+	var keyspaces []string
+	if err := json.Unmarshal(body, &keyspaces); err != nil {
+		return nil, err
+	}
+	return keyspaces, nil
+}
+
+// ListKeyspaces calls the management API to list existing keyspaces
+func (client *NodeMgmtClient) ListKeyspaces(pod *corev1.Pod, keyspaceName string) ([]string, error) {
```
Comment (burmanm) [inline pkg/httphelper/client.go]: Can there be multiple keyspaces with the same name? ;) Or why is "ListKeyspaces" requiring keyspaceName to list?

Comment (adejanovski) [reply pkg/httphelper/client.go]: It was a way of filtering on the management API side if you're looking for a single keyspace and don't want to filter on the client side.
Dutra seemed to like the idea of doing this (or maybe it was Erik, can't remember), and I'm fine with it 🙂

Comment (burmanm) [reply pkg/httphelper/client.go]: In that case, shouldn't it be called "GetKeyspace" ? Or how does one List those keyspaces?

Comment (adejanovski) [reply pkg/httphelper/client.go]: Leaving it empty will list all keyspaces: https://github.com/k8ssandra/management-api-for-apache-cassandra/commit/e77aa6ab32fb119fbb865cf8edc3f8a9a37eab95#diff-51532ffcd7720e83141217c410cb08e76cf95fd93e2ad354abd72b71c78aa8d4R174

Comment (burmanm) [reply pkg/httphelper/client.go]: Indeed, that request in the Test removes the parameter:

```java
MockHttpRequest request = MockHttpRequest.get(ROOT_PATH + "/ops/keyspace");
```

This implementation always adds the parameter keyspaceName (which is called filter in those Java tests). Thus, the implementation here should add the filter only ``if keyspaceName != ""`` 

```go
request := nodeMgmtRequest{
		endpoint: fmt.Sprintf("/api/v0/ops/keyspace?keyspaceName=%s", keyspaceName),
```

Comment (adejanovski) [reply pkg/httphelper/client.go]: But it'll work either way as passing an empty value for `keyspaceName` will give the same result as not passing it.
Would you rather have two distinct functions here, one with the `keyspaceName` arg and the other without?

Comment (burmanm) [reply pkg/httphelper/client.go]: You can't do that in Golang without naming them. What guarantees would the reader have that keyspaceName empty is same as not setting it? If even the management-api has no guarantees or test for that?

**Review by burmanm — APPROVED** [review]

**Pull request merged**